src/prefect/_internal/schemas/validators.py

```diff
@@ -100,7 +112,9 @@ def validate_values_conform_to_schema(
         ) from exc
 
 
-def infrastructure_must_have_capabilities(value):
+def infrastructure_must_have_capabilities(
+    value: Union[Dict[str, Any], "Block", None],
+) -> Optional["Block"]:
```

Comment:
This PR adds more type hinting to our validators module, since this is in line with _why we are gathered here today_